death-causes-app/src/components/Question.tsx

```diff
@@ -0,0 +1,409 @@
+import React, { ChangeEvent, PureComponent, ReactElement } from "react";
+import "./QuestionMenu.css";
+import Button from "react-bootstrap/Button";
+import Popover from "react-bootstrap/Popover";
+import OverlayTrigger from "react-bootstrap/OverlayTrigger";
+import InputGroup from "react-bootstrap/InputGroup";
+import { Row, Col, Form } from "react-bootstrap";
+import "./Question.css";
+import MarkDown from "react-markdown";
+import { InputValidity } from "../models/Factors";
+import DropdownButton from "react-bootstrap/DropdownButton";
+import Dropdown from "react-bootstrap/Dropdown";
+import UnitPicker from "./UnitPicker";
+import { DropdownToggle } from "reactstrap";
+
+const BACKGROUNDCOLOR_DISABLED = "#c7c7c7";
+const TEXTCOLOR_DISABLED = "#999";
+const BACKGROUNDCOLOR_CHOICE = "#cef1f5";
+const ERROR_COLOR = "#fc0303";
+const WARNING_COLOR = "#bfa50d";
+
+interface I_Question<T> {
+  handleChange: (e: ChangeEvent<HTMLInputElement>) => void;
+  handleIgnoreFactor: (factorname: string) => void;
+  name: string;
+  phrasing: string;
+  factorAnswer: T;
+  helpText: string;
+  featured: boolean;
+}
+
+interface I_QuestionStates {
+  ignore: boolean;
+}
+
+interface NumericQuestionProps extends I_Question<number> {
+  placeholder: string;
+  inputvalidity: InputValidity;
+  unitOptions: string[];
+  handleChangeUnit: (fname: string, newUnitName: string) => void;
+}
+
+interface StringQuestionProps extends I_Question<string> {
+  placeholder: string;
+  inputvalidity: InputValidity;
+  options: string[];
+}
+
+interface FormControlStyle {
+  [key: string]: string;
+}
+
+interface AbstractQuestionProps {
+  name: string;
+  ignore: boolean;
+  handleIgnoreBox: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  helpText: string;
+  phrasing: string;
+  secondLine: ReactElement | string;
+  featured: boolean;
+  unitText: string | React.ReactNode;
+}
+
+class Question extends React.PureComponent<AbstractQuestionProps> {
```

Comment:
Jeg synes det er `AbstractQuestionProps` lyder som om Question klassen er abstract og dermed ikke kan instantieres, men alligevel renderer både string og number question et Question objekt. Det virker lidt som om at vi tænker at number og string questions er så ens at vi har brug for en fælles overordnet klasse, men alligevel er der meget specifik funktionalitet til hver af klasserne, så måske er de virkeligheden ikke så ens at vi har brug en generel Question klasse? Du kan overveje det - det er i hvert fald lidt forvirrende nu synes jeg.

Reply:
Det er fordi AbstractQuestion slet ikke er en abstrakt klasse men bare en ReactComponent. Jeg omdøber den til QuestionContext fordi den indeholder netop context til inputfeltet(som enten er et tekstfelt eller en dropdown)